packages/relay/tests/helpers.ts

```diff
@@ -885,40 +885,98 @@ export const calculateTxRecordChargeAmount = (exchangeRateIncents: number) => {
 };
 
 export const useInMemoryRedisServer = (logger: Logger, port: number) => {
-  let envsToReset: { TEST?: string; REDIS_ENABLED?: string; REDIS_URL?: string };
+  overrideEnvs({ TEST: 'false', REDIS_ENABLED: 'true', REDIS_URL: `redis://127.0.0.1:${port}` });
+
   let redisInMemoryServer: RedisInMemoryServer;
 
   before(async () => {
-    ({ envsToReset, redisInMemoryServer } = await startRedisInMemoryServer(logger, port));
+    redisInMemoryServer = await startRedisInMemoryServer(logger, port);
   });
 
   after(async () => {
-    await stopRedisInMemoryServer(redisInMemoryServer, envsToReset);
+    await stopRedisInMemoryServer(redisInMemoryServer);
   });
 };
 
 export const startRedisInMemoryServer = async (logger: Logger, port: number) => {
   const redisInMemoryServer = new RedisInMemoryServer(logger.child({ name: 'RedisInMemoryServer' }), port);
   await redisInMemoryServer.start();
-  const envsToReset = {
-    TEST: process.env.TEST,
-    REDIS_ENABLED: process.env.REDIS_ENABLED,
-    REDIS_URL: process.env.REDIS_URL,
-  };
-  process.env.TEST = 'false';
-  process.env.REDIS_ENABLED = 'true';
-  process.env.REDIS_URL = `redis://127.0.0.1:${port}`;
-  return { redisInMemoryServer, envsToReset };
+  return redisInMemoryServer;
 };
 
-export const stopRedisInMemoryServer = async (
-  redisInMemoryServer: RedisInMemoryServer,
-  envsToReset: { TEST?: string; REDIS_ENABLED?: string; REDIS_URL?: string },
-): Promise<void> => {
+export const stopRedisInMemoryServer = async (redisInMemoryServer: RedisInMemoryServer): Promise<void> => {
   await redisInMemoryServer.stop();
-  process.env.TEST = envsToReset.TEST;
-  process.env.REDIS_ENABLED = envsToReset.REDIS_ENABLED;
-  process.env.REDIS_URL = envsToReset.REDIS_URL;
+};
+
+/**
+ * Temporarily overrides environment variables for the duration of the encapsulating describe block.
+ * @param envs - An object containing key-value pairs of environment variables to set.
+ *
+ * @example
+ * describe('given TEST is set to false', () => {
+ *   overrideEnvs({ TEST: 'false' });
+ *
+ *   it('should return false', () => {
+ *     expect(process.env.TEST).to.equal('false');
+ *   });
+ * });
+ *
+ * it('should return true', () => {
+ *   expect(process.env.TEST).to.equal('true');
+ * });
+ */
+export const overrideEnvs = (envs: NodeJS.Dict<string>) => {
```

Comment:
Q: should we make the name more descriptive?
Something like `overrideEnvsInJestDescribe` and `withOverriddenEnvsInJestTest` or better names.
If these methods were coming from the jest library or were overriding classes it'd be fine to be implicit but used outside it can be missed.

My main concern here is engineering time spent writing and debugging tests missing how the env variable in the class are affecting the tests scope.
Seeing with clarity that the use of these methods has an impact in relation to jest scoping might avoid issues.

Reply:
Done, renamed to `overrideEnvsInMochaDescribe` and `withOverriddenEnvsInMochaTest`.